uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/calendar-feeds/link-modal/input-with-copy-button.directive.js

```diff
@@ -0,0 +1,42 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash'
+], function (_) {
+  InputWithCopyButton.__name = 'inputWithCopyButton';
+  InputWithCopyButton.$inject = ['shared-settings'];
+
+  return InputWithCopyButton;
+
+  function InputWithCopyButton (sharedSettings) {
+    var templateUrl = sharedSettings.sourcePath + 'calendar-feeds/link-modal/input-with-copy-button.html';
+
+    return {
+      controller: _.noop,
+      controllerAs: 'input',
+      link: InputWithCopyButtonLink,
+      restrict: 'E',
+      scope: {},
+      templateUrl: templateUrl,
+      require: {
+        model: '^ngModel'
+      }
+    };
+
+    function InputWithCopyButtonLink ($scope, $element, $attr, $ctrl) {
+      var vm = $scope.input;
+
+      vm.model = $ctrl.model;
+
+      vm.copy = copy;
```

Comment:
not sure the space above is needed

Reply:
Yes it is, the first is a property definition, the second is a method definition.